### PR TITLE
update for Vite

### PR DESCRIPTION
**Pull request by AlexErrant**
Hi, thanks for a great library. This PR updates the Readme with instructions for Vite and adds Vite to the test suite.

---

_Without_ `@surma/rollup-plugin-off-main-thread`, the `vite build` command fails with

```
❯ npx vite build
vite v5.1.4 building for production...
transforming (1) index.html
<--- Last few GCs --->

[517549:0x572c270]    75048 ms: Mark-sweep 4041.1 (4138.2) -> 4030.7 (4141.4) MB, 1570.5 / 0.0 ms  (average mu = 0.195, current mu = 0.041) task; scavenge might not succeed
[517549:0x572c270]    77747 ms: Mark-sweep 4045.6 (4142.2) -> 4034.5 (4145.2) MB, 2632.7 / 0.0 ms  (average mu = 0.094, current mu = 0.025) task; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb7a940 node::Abort() [node]
 2: 0xa8e823  [node]
 3: 0xd5c940 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xd5cce7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xf3a3e5  [node]
 6: 0xf4c8cd v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 7: 0xfbb924 v8::internal::ScavengeJob::Task::RunInternal() [node]
 8: 0xe2c07b non-virtual thunk to v8::internal::CancelableTask::Run() [node]
 9: 0xbe6884  [node]
10: 0xbe9cee node::PerIsolatePlatformData::FlushForegroundTasksInternal() [node]
11: 0x165eb16  [node]
12: 0x1671054  [node]
13: 0x165f47e uv_run [node]
14: 0xabda6d node::SpinEventLoop(node::Environment*) [node]
15: 0xbc1164 node::NodeMainInstance::Run() [node]
16: 0xb35bc8 node::LoadSnapshotDataAndRun(node::SnapshotData const**, node::InitializationResult const*) [node]
17: 0xb3976f node::Start(int, char**) [node]
18: 0x7f46b3fb0083 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
19: 0xabbdee _start [node]
Aborted
```

However, _with_ `@surma/rollup-plugin-off-main-thread`, the dev server `npx vite dev` causes the page to try to load  `http://localhost:5173/pkg/snippets/wasm-bindgen-rayon-3e04391371ad0a8e/src/undefined` which obviously doesn't exist and results in `Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec.` [Possibly related.](https://github.com/vitejs/vite/issues/1820)

Therefore, we conditionally use `@surma/rollup-plugin-off-main-thread` based on Vite's "mode".

### CODE REVIEW

**Comment by AlexErrant** [comment]
On [a separate but related PR](https://github.com/open-spaced-repetition/fsrs-browser/pull/11), I noticed that `vite build` fails with the same "JavaScript heap out of memory" error when using wasm-bindgen-rayon in a separate webworker ([as was mandatory in <=v1.1.3](https://github.com/RReverser/wasm-bindgen-rayon/blob/v1.1.3/README.md#:~:text=You%20must%20instantiate%20the%20main%20JS%2BWasm%20in%20a%20dedicated%20Worker%20to%20avoid%20blocking%20the%20main%20thread)) even when using `@surma/rollup-plugin-off-main-thread`. I had to [move the code out of the web worker](https://github.com/open-spaced-repetition/fsrs-browser/pull/11/commits/e9edc18d9e1562e5d7d0c8d2ae5c9f4fb80fa7de) to get `vite build` to work.

Hypothetically, if I _wanted_ to use a separate web worker, would you know the solution to this? Don't waste too much time if you don't know - I really don't mind not having to spin up another worker. Just asking for the sake of completeness :)

**Comment by JohannesKlauss** [comment]
Sorry to chime in here, but I have this issue as well, though using `@surma/rollup-plugin-off-main-thread` won't fix it. The build is successful, but vite doesn't bundle all the needed files (i.e. the wasm file is missing in the output, also the actual worker file).
I followed your vite.config but this doesn't seem to do the trick for me. As far as I understand it the out of memory exception happens because of the circular dependency between main.js -> workerHelpers.js -> workerHelpers.worker.js

@AlexErrant Can you give me any pointer on why the build isn't including the needed files?

**Comment by AlexErrant** [comment]
Updating Vite to 5.1.6+ removes the need for `@surma/rollup-plugin-off-main-thread`. This PR now simply adds a test, which may be useful to guard against regressions.

**Comment by 9SMTM6** [comment]
@RReverser since this only adds a test case at this point, it should be unproblematic to rebase and merge.

File: test/vite.config.mjs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { defineConfig } from 'vite';
+
+export default defineConfig(() => {
+	return {
+		server: {
+			headers: {
+				'Cross-Origin-Opener-Policy': 'same-origin',
+				'Cross-Origin-Embedder-Policy': 'require-corp',
+			},
```
Comment (RReverser) [inline test/vite.config.mjs:9]: Do we need these? We're not using Vite server after this PR, right?

Comment (AlexErrant) [reply test/vite.config.mjs]: Ah yeah; removed!

**Comment by RReverser** [comment]
Thanks!

**Pull request merged**